README.md

```diff
@@ -1,170 +1,189 @@
+
+First, clone the project from GitHub:
 
-### 2. Running Locally
+```bash
+git clone <repository-url>
+cd <repository-directory>
+````
 
-#### Option 1: Development Mode
+### 2\. Install Dependencies
 
-Run backend and frontend separately for hot reload:
+Install system-level and Python dependencies.
 
 ```bash
-# Terminal 1: Start the FastAPI backend
-uv run fastapi dev src/agent_monitor/main.py
+# Install system dependencies (pre-commit, go-task, tilt) using Homebrew
+brew bundle
 
-# Terminal 2: Start the React frontend with proxy
-cd frontend && npm run dev
+# Install Python packages using uv
+uv sync
 ```
 
-The application will be available at:
-- **Frontend UI**: http://localhost:3000/ (with hot reload)
-- **Backend API**: http://localhost:8000/ (API calls proxied from frontend)
-- **API Documentation**: http://localhost:8000/docs
+### 3\. Run the Application
 
-#### Option 2: Production Mode (Single container)
+You can run the service directly with `uv` or in a Kubernetes cluster using `tilt`.
 
-Build frontend and serve everything from FastAPI:
+**Option A: Locally with `uv`**
 
-```bash
-# Build the frontend
-cd frontend && npm run build && cd ..
+This is the simplest way to get started.
 
-# Start the integrated server
+```bash
+# Start a development server with hot-reloading
 uv run fastapi dev src/agent_monitor/main.py
 ```
+```bash
+# Or, start a production-like server
+uv run fastapi run src/agent_monitor/main.py
+```
+Once running, you can access the following endpoints:
 
-The application will be available at:
-- **Frontend UI**: http://localhost:8000/
-- **API Documentation**: http://localhost:8000/docs
+  - **API Server**: `http://localhost:8000`
+  - **Health Check**: `http://localhost:8000/health`
+  - **API Docs (Swagger)**: `http://localhost:8000/docs`
+
+**Option B: With Kubernetes with Tilt**
+
+This method is ideal for simulating a production environment.
 
-#### Option 3: Kubernetes with Tilt
 ```bash
-# From project root:
+# Ensure your local Kubernetes cluster is running, then start tilt
 tilt up
-
-# The app will be available at http://localhost:10005
-# Tilt UI available at http://localhost:10350
 ```
+The application will be available at `http://localhost:10005`, with the Tilt UI at `http://localhost:10350`.
 
-### 3. Code Quality
 
-```bash
-uv run poe check      # Run all quality checks
-uv run poe mypy       # Type checking
-uv run poe ruff       # Linting and formatting
-uv run poe bandit     # Security analysis
-```
+-----
+
+## Configuration
+
+Current configurations are managed within the source files.
+
+Key configuration points include:
 
-## Local Kubernetes Setup
+  * **Server Host/Port**: Defined in the FastAPI application startup logic in `src/agent_monitor/main.py`.
+  * **Logging**: Configured in `src/agent_monitor/main.py`, with health check endpoints excluded from access logs.
+  * **OTLP Span Processing**: Business logic is located in `src/agent_monitor/core/span_preprocessor.py`.
```

Comment:
This does not fit here. I tend to remove the whole section, as there is nothing really useful here for the user.